src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -383,9 +383,19 @@ CHIP_ERROR Browse(void * context, DnssdBrowseCallback callback, uint32_t interfa
     std::string regtype(type);
     std::string subtypeDelimiter = "._sub.";
     size_t position              = regtype.find(subtypeDelimiter);
+    // if we have more in the string (a subtype) then append it to the end
     if (position != std::string::npos)
     {
-        regtype = regtype.substr(position + subtypeDelimiter.size()) + "," + regtype.substr(0, position);
+        if (regtype.rfind("_", position) == 0)
+        {
+            regtype = regtype.substr(position + subtypeDelimiter.size()) + "," + regtype.substr(0, position);
+        }
+        else
+        {
+            // if it doesn't start with "_" then don't include the instance name
```

Comment:
This needs a much clearer explanation of why (the current comment just describes "what", which is already clear from the code... though even on the "what" the code does not seem to match the comment).  What cases is this trying to handle and why is this the right way to handle them?

Reply:
Updated the comment with more explanation and examples